terraform-provider-azurerm_v3.49.0_x5.exe plugin crashed

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

<!--- Please keep this note for the community --->

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

<!--- Thank you for keeping this note for the community --->


### Terraform Version

1.4.2

### AzureRM Provider Version

3.49.0

### Affected Resource(s)/Data Source(s)

azurerm_windows_web_app

### Terraform Configuration Files

```hcl
resource "azurerm_windows_web_app" "example" {
  name                = "${var.prefix}-${var.environment}-${var.suffix}"
  location            = var.location
  resource_group_name = var.rg_name
  service_plan_id     = var.plan_id

  identity {
    type = "SystemAssigned"
  }

  auth_settings_v2 {
    auth_enabled           = var.auth_enabled
    default_provider       = "aad"
    unauthenticated_action = "RedirectToLoginPage"
    require_authentication = true
    require_https          = true

    active_directory_v2 {
      client_id                  = var.aad_client_id
      allowed_audiences          = var.aad_allowed_audiences
      tenant_auth_endpoint       = "https://sts.windows.net/<tenant_id>/"
      client_secret_setting_name = "MICROSOFT_PROVIDER_AUTHENTICATION_SECRET"
    }

    login {}
  }

  site_config {
    always_on              = true
    default_documents      = ["index.html"]
    ftps_state             = "FtpsOnly"
    vnet_route_all_enabled = true
    use_32_bit_worker      = false
    cors {
      allowed_origins     = var.cors_allowed_origins
      support_credentials = true
    }
  }

  app_settings = {
    "MICROSOFT_PROVIDER_AUTHENTICATION_SECRET" = "@Microsoft.KeyVault(VaultName=${var.vault_name};SecretName=${var.client_secret_name})"
  }
}
```


### Debug Output/Panic Output

```shell
│ Error: Plugin did not respond
│
│   with module.api-app.azurerm_windows_web_app.example,
│   on [..]\main.tf line 1, in resource "azurerm_windows_web_app" "example":
│    1: resource "azurerm_windows_web_app" "example" {
│
│ The plugin encountered an error, and failed to respond to the plugin.(*GRPCProvider).ApplyResourceChange call. The
│ plugin logs may contain more details.
╵
Stack trace from the terraform-provider-azurerm_v3.49.0_x5.exe plugin:

panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xc0000005 code=0x0 addr=0x0 pc=0x46426a6]

goroutine 121 [running]:
github.com/hashicorp/terraform-provider-azurerm/internal/services/appservice/helpers.expandAadAuthV2Settings({0xc0017cd420, 0x1, 0x0?})
        github.com/hashicorp/terraform-provider-azurerm/internal/services/appservice/helpers/auth_v2_schema.go:988 +0x606
github.com/hashicorp/terraform-provider-azurerm/internal/services/appservice/helpers.ExpandAuthV2Settings({0xc002940680, 0x1, 0x7257e80?})
        github.com/hashicorp/terraform-provider-azurerm/internal/services/appservice/helpers/auth_v2_schema.go:2059 +0x12a
github.com/hashicorp/terraform-provider-azurerm/internal/services/appservice.WindowsWebAppResource.Update.func1({0x7265408, 0xc002fc8960}, {0xc001ed1c00, {0x7265f30, 0xc0006c2990}, 0xc0026ae600, 0x0, {0x7267318, 0xb7f6a48}})
        github.com/hashicorp/terraform-provider-azurerm/internal/services/appservice/windows_web_app_resource.go:847 +0x1cc5
github.com/hashicorp/terraform-provider-azurerm/internal/sdk.(*ResourceWrapper).Resource.func7({0x7265408, 0xc002fc8960}, 0x1a3185c5000?, {0x5e19e60?, 0xc001ed1c00?})
        github.com/hashicorp/terraform-provider-azurerm/internal/sdk/wrapper_resource.go:116 +0x15c
github.com/hashicorp/terraform-provider-azurerm/internal/sdk.diagnosticsWrapper.func1({0x7265408, 0xc002fc8960}, 0x0?, {0x5e19e60, 0xc001ed1c00})
        github.com/hashicorp/terraform-provider-azurerm/internal/sdk/wrapper_resource.go:185 +0xbe
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).update(0xc0006f7c00, {0x7265440, 0xc0025e12c0}, 0xd?, {0x5e19e60, 0xc001ed1c00})
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.24.1/helper/schema/resource.go:741 +0x12e
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).Apply(0xc0006f7c00, {0x7265440, 0xc0025e12c0}, 0xc00219f040, 0xc0026ae480, {0x5e19e60, 0xc001ed1c00})
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.24.1/helper/schema/resource.go:847 +0x83a
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ApplyResourceChange(0xc0003846c0, {0x7265440?, 0xc0025e11a0?}, 0xc0025f0410)
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.24.1/helper/schema/grpc_provider.go:1021 +0xe8d
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ApplyResourceChange(0xc000bff540, {0x7265440?, 0xc0025e0990?}, 0xc00065ccb0)
        github.com/hashicorp/terraform-plugin-go@v0.14.3/tfprotov5/tf5server/server.go:818 +0x574
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ApplyResourceChange_Handler({0x656d100?, 0xc000bff540}, {0x7265440, 0xc0025e0990}, 0xc00065cc40, 0x0)
        github.com/hashicorp/terraform-plugin-go@v0.14.3/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:385 +0x170
google.golang.org/grpc.(*Server).processUnaryRPC(0xc0001f8000, {0x7275660, 0xc001330820}, 0xc0025e8c60, 0xc000b92930, 0xb790d40, 0x0)
        google.golang.org/grpc@v1.51.0/server.go:1340 +0xd23
google.golang.org/grpc.(*Server).handleStream(0xc0001f8000, {0x7275660, 0xc001330820}, 0xc0025e8c60, 0x0)
        google.golang.org/grpc@v1.51.0/server.go:1713 +0xa2f
google.golang.org/grpc.(*Server).serveStreams.func1.2()
        google.golang.org/grpc@v1.51.0/server.go:965 +0x98
created by google.golang.org/grpc.(*Server).serveStreams.func1
        google.golang.org/grpc@v1.51.0/server.go:963 +0x28a

Error: The terraform-provider-azurerm_v3.49.0_x5.exe plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.
```


### Expected Behaviour

_No response_

### Actual Behaviour

_No response_

### Steps to Reproduce

`terraform apply`

### Important Factoids

_No response_

### References

Was glad to see #21091 being released in 3.49.0 of the provider but it seems it's still not working.

--- a/internal/services/appservice/helpers/auth_v2_schema.go
+++ b/internal/services/appservice/helpers/auth_v2_schema.go
@@ -985,7 +985,7 @@ func expandAadAuthV2Settings(input []AadAuthV2Settings) *web.AzureActiveDirector
 			}
 		}
 		if len(aad.AllowedAudiences) > 0 {
-			result.Validation.AllowedAudiences = pointer.To(aad.AllowedIdentities)
+			result.Validation.AllowedAudiences = pointer.To(aad.AllowedAudiences)
 		}
 	}
 

